ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,664 @@
+import itertools
+import warnings
+from collections import defaultdict
+from typing import Any, Callable, cast, Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+from typing_extensions import Literal
+
+import ignite.distributed as idist
+from ignite.distributed.utils import _all_gather_tensors_with_shapes
+from ignite.metrics.metric import reinit__is_reduced
+from ignite.metrics.recall import _BasePrecisionRecall
+from ignite.utils import to_onehot
+
+
+class MeanAveragePrecision(_BasePrecisionRecall):
+    _tp: Dict[int, List[torch.Tensor]]
+    _fp: Dict[int, List[torch.Tensor]]
+    _scores: Union[Dict[int, List[torch.Tensor]], List[torch.Tensor]]
+    _P: Union[Dict[int, int], List[torch.Tensor]]
+
+    def __init__(
+        self,
+        rec_thresholds: Optional[Union[Sequence[float], torch.Tensor]] = None,
+        average_operand: Optional[Literal["precision", "max-precision"]] = "precision",
+        class_mean: Optional[Literal["micro", "macro", "weighted", "with_other_dims"]] = "macro",
+        classification_is_multilabel: bool = False,
```

Comment:
```suggestion
        is_multilabel: bool = False,
```
Let's not give all details in the arg name otherwise it is too many chars in the name

Reply:
I added "classification" since it's only used in classification task. But if you think the arg description is enough, I change it.

Reply:
Yes, keep arg name same as in `_BasePrecisionRecall`: `is_multilabel`